router/cmd/main.go

```diff
@@ -165,10 +181,28 @@ func Main() {
 		logger.Error("Could not shutdown server", zap.Error(err))
 	}
 
+	if *cpuprofile != "" {
```

Comment:
I think we can defer this from line ~44

Reply:
no, the shutdown is too fast
it needs to run before we return